### ISSUE TITLE
Missing signatures

### ISSUE THREAD

**Issue by wargio** [rizinorg/sigdb#1]
# Bin signatures

## PE
- [x] https://docs.microsoft.com/en-us/cpp/c-runtime-library/crt-library-features?view=msvc-170#c-runtime-lib-files
- [x] https://docs.microsoft.com/en-us/cpp/c-runtime-library/crt-library-features?view=msvc-170#c-standard-library-lib-files
- [x] https://www.masm32.com/

### Ubuntu
 - [x] LLVM C++ Standard library (release 13, 12, 11, 10, 9, 8, 7)
 - [x] GNU C Library
 - [x] libgcc (release 11, 10, 7, 8)
 - [x] libstdc++ v3 (release 5, 6, 7, 8)
 - [x] musl C standard library
 - [x] OpenSSL (libssl + libcrypto)
 - [x] Linux seccomp filter
 - [x] Libsodium
 - [x] zlib

### Debian
 - [ ] LLVM C++ Standard library (release 16, 15, 14, 13, 12, 11, 10, 9, 8, 7)
 - [ ] GNU C Library
 - [ ] libgcc (release 11, 10, 7, 8)
 - [ ] libstdc++ v3 (release 5, 6, 7, 8)
 - [ ] musl C standard library
 - [ ] OpenSSL (libssl + libcrypto)
 - [ ] Linux seccomp filter
 - [ ] Libsodium

## MACHO
- [ ] All static libs available for M1 and x86 on XCode

# Specific signatures

#### Windows

 - [ ] Windows Driver Kit (arm, x86, x64)
 - [x] Visual Studio 2005, 2008, 2010, 2012, 2013, 2015, 2017, 2019
 - [x] Visual Studio 2022
 - [ ] MFC 6.0, 7.1, 8.0, 9.0 +++ (arm, x86, x64)

#### Android

 - [x] Android NDK (arm, mips, x86)
 - [ ] Linux kernel? (arm, mips, x86)

#### Apple IOS/OSX (mach0)

 - [ ] [iPhoneOS SDK](https://github.com/xybp888/iOS-SDKs) (arm)
 - [ ] xnu (arm, ppc, x86, x64)
 - [ ] mac os sdk (arm, x86, x64)
 - [ ] Objective-C ? (arm, x86, x64)
 - [ ] Swift ? (arm, x86, x64)

#### Consoles

 - [ ] Playstation 1 SDK/PsyQ? (mips)
 - [ ] PSP SDK? (mips)

**Comment by segevfiner** [comment]
P.S. When FLIRT was originally implemented I think @williballenthin mentioned https://github.com/mandiant/siglib/ which generates/includes FLIRT signatures for MSVC runtime libraries.

**Comment by wargio** [comment]
yes, but for that you need IDA tools. but the repo is interesting because it has the sources, which i can use here. thanks!!
if you use rizin on dev, you can actually generate FLIRT signatures, the feature will be publicly announced on 0.4.0 release.

**Comment by wargio** [comment]
I have added Visual Studio 2005, 2008, 2010, 2012, 2013 and 2015 for arm32, mips32, x86 and x64 in the repo.

**Comment by segevfiner** [comment]
So missing 2017, 2019 & 2022. Might want to note this in the issue description.

**Comment by segevfiner** [comment]
BTW. I think that repo has 2017 & 2019 and also ATL/MFC. It grabberd those using Docker on Windows, so the Dockerfile there can be modified to switch to the VS 2022 build tools installer to grab those as well. (You need a Windows machine for Docker for Windows, there are freely available VMs from Microsoft: https://developer.microsoft.com/he-il/windows/downloads/virtual-machines/)

**Comment by wargio** [comment]
the whole windows 10/11 SDK is already in the repo. the only thing missing is mostly the libs included in VS 17/19/22

**Comment by williballenthin** [comment]
Which libs are missing from 2017/2019? I may still have a setup that I can use to pull the files (and update siglib). 

Maintaining these signature sets is a bit of a pain, so I’d love to collaborate and have this done once for everyone. Please don’t hesitate to take sigs from siglib if they’re useful. 

2017 files are here for example: https://github.com/mandiant/siglib/tree/master/data/VS2017

**Comment by segevfiner** [comment]
I'm not sure if any are missing from 2017 & 2019, barring of course the fact that they have many minor versions of both the SDK, CRT, & UCRT, and the repo likely only includes a specific one. 2022 is missing (This repo was created before it existed AFAIK).

**Comment by wargio** [comment]
according to this https://developer.microsoft.com/en-us/windows/downloads/sdk-archive/ i have them all + the ones included with old VS code.
![image](https://user-images.githubusercontent.com/561184/154059551-83e574d2-c5ea-482b-95aa-2afa96033279.png)

![image](https://user-images.githubusercontent.com/561184/154059713-cfd43a51-5770-49dd-a27d-0a9db7e4d95b.png)

**Comment by wargio** [comment]
oh, the sources are here: https://github.com/rizinorg/sigdb-source/tree/main/pe/x86/64/winsdk

**Comment by segevfiner** [comment]
The winsdk folder ones seem to have the ucrt, I was talking about MSVC 2017 & 2019 CRT which also have more minor versions past a certain version of Visual Studio where they started updating more frequently AFAIK:
* https://github.com/mandiant/siglib/tree/master/data/VS2017
* https://github.com/mandiant/siglib/tree/master/data/VS2019

MSVC 2015 and above still have their own version specific CRT built atop the UCRT which allows them to interop.

**Comment by wargio** [comment]
then i did not add it yet, as stated above

**Comment by k-badz** [comment]
Hello everyone, do you have any rough estimate when MSVC 2017+ signatures would be added? Cutter doesn't recognize some of the basic stuff from new compilers:

Ghidra:
![image](https://user-images.githubusercontent.com/101342783/211210705-8a932b53-4e84-49d5-9c31-75a9d5abb558.png)

Cutter:
![image](https://user-images.githubusercontent.com/101342783/211210730-eee2b7b8-c5b7-48ab-bfad-f01f20d5dba6.png)

Thanks in advance!

**Comment by wargio** [comment]
tbh, i haven't had chance to download the new VC versions.

**Comment by wargio** [comment]
I have added VC 2017 and VC 2019 to the db https://github.com/rizinorg/sigdb/commit/d990637aec283048adb0ede336a70aa513de4eb3

**Comment by k-badz** [comment]
Great, thanks a lot

**Comment by wargio** [comment]
VC 2022 was added https://github.com/rizinorg/sigdb/commit/829baf835e3515923266898fd597f7f75046ebd2
